docs/windows-design.md

```diff
@@ -0,0 +1,249 @@
+# Running Antrea on Windows
+
+Antrea supports running on Windows worker Node. In this case, the Windows Node should
+be a worker in the cluster, while the master Node must be Linux. On Windows Node, 
+Antrea sets up an overlay network to forward packets between Nodes and implements
+NetworkPolicies. 
+
+## Design
+
+On Windows, the Host Networking Service (HNS) is a necessary component to support container
+networking. For Antrea on Windows, “Transparent” mode is chosen for the HNS network. In this
+mode, containers will be directly connected to the physical network through an **external** Hyper-V switch. 
+
+OVS is working as the extension of the external Hyper-V switch, which is bound with the HNS 
+Network. Hence, the packets that are sent from/to the containers could be processed by OVS.
+The network adapter used in the HNS Network is also added into the OVS bridge as the uplink
+interface. An interface for the OVS bridge is created, and the original networking
+configuration(e.g., IP, MAC and route entries) on the host network adapter are moved to
+this new interface. Some extra OpenFlow entries are needed to ensure the host traffic can be
+forwarded correctly.
+
+<img src="/docs/assets/hns_integration.svg" width="600" alt="HNS Integration">
+
+SNAT based on OpenFlow is needed to make sure the containers could access the external address.
+The SNATed address is using the IP configured on the OVS bridge. Some assisted OpenFlow entries are
+installed to help identify and forward the external traffic.
+
+kube-proxy is running on each Windows Node to provide Kubernetes Service. The Service traffic is
+output to the host from gw0, and captured by kube-proxy. kube-proxy sets up a new connection to
+the backend endpoint, and forwarded the request using this new connection. The forwarded request
+enters OVS pipeline again and output to the actual Pod. To be compatible with OVS, kube-proxy on
+Windows must be configured to run in the **userspace** mode, and a specific network adapter is 
+required to be provided on which the Service IP address is configured.
+
+### HNS Network configuration
+
+HNS Network is created at Antrea Agent initialization phase, and it should be created before creating 
+OVS bridge. This is because OVS is working as the Hyper-V Switch Extension, and the ovs-vswitchd process
```

Comment:
> the ovs-vswitchd process cannot run until the Hyper-V Switch is created

This is confusing to me because it seems `PrepareNode.ps1` takes care of starting ovs-vswitchd (right after installing OVS) and that happens before the HNS network is created?

https://github.com/vmware-tanzu/antrea/blob/master/hack/windows/Install-OVS.ps1#L161